Add failing test for #6314  and PR #6356

astepin opened the pull request:
This is a failing test for issue #6314 in addition to fix in PR #6356 .

gander committed:
Fix wrong variable name

Fix $fieldMapping to $embeddedClass in generateEmbeddedPropertyDocBlock

astepin committed:
Add failing test for gh-#6314  (gh-#6356)

Ocramius commented on tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php:
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
+
+        $docComment = $refClass->getProperty('testEmbedded')->getDocComment();
+
+        $needle = sprintf('@Embedded(class="%s", columnPrefix=\'%s\')', $testMetadata->name,'prefix');
```
Fairly sure that we don't use single quotes on any annotations (see `columnPrefix`)

Ocramius commented on tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php:
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
+        $docComment = $refClass->getProperty('testEmbedded')->getDocComment();
+
+        $needle = sprintf('@Embedded(class="%s", columnPrefix=\'%s\')', $testMetadata->name,'prefix');
+        self::assertContains($needle, $docComment);
```
I see [from the test runs](https://travis-ci.org/doctrine/doctrine2/jobs/240964836#L342) that this is the only failing bit: a fix should be easy to provide now :-)

Ocramius commented on tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php:
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
+
+        $needle = sprintf('@Embedded(class="%s", columnPrefix=\'%s\')', $testMetadata->name,'prefix');
+        self::assertContains($needle, $docComment);
+
```
Remove this empty line

Ocramius commented on tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php:
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
+
+        self::assertTrue($refClass->hasProperty('testEmbedded'));
+
+
```
Remove this empty line

Ocramius commented on tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php:
```diff
@@ -224,6 +224,26 @@ public function newInstance($metadata)
+    public function testEmbeddedEntityWithColumnPrefix()
+    {
+        $testMetadata = $this->generateTestEmbeddableFixture();
+        $isbnMetadata = $this->generateIsbnEmbeddableFixture(['testEmbedded' => $testMetadata], 'prefix');
```
Please use a random value for the prefix passed in here - makes it harder to escape checks

astepin committed:
Merge branch 'original-pr-6356' into gh-6356

astepin committed:
- fixed columnPrefix generator for the embedded annotation

astepin commented:
I had to merge PR #6356 to my PR in order to solve another issue i've discovered. 
As described [here](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/embeddables.html#column-prefixing), columnPrefix could be a string or a boolean.This has an impact on the generation of this annotation. I've fixed this and added appropriet tests.

Ocramius commented:
LGTM :+1:

Ocramius commented:
Thanks @astepin, sorry for the late feedback :-)
